src/main/java/seedu/duke/operationlist/FlightList.java

```diff
@@ -0,0 +1,344 @@
+    protected static final int CHECK_MISSING_DETAIL = 0;
+    protected static final int AIRLINE_LENGTH_LIMIT = 22;
+    protected static final int DESTINATION_LENGTH_LIMIT = 22;
+    protected static boolean isFlightNumberPresent = false;
+    protected static boolean isDepartureTimePresent = false;
+    protected static boolean isGateNumberPresent = false;
+    protected static String timeRegex = "([01]?[0-9]|2[0-3])[0-5][0-9]";
+    protected static String checkInRegex = "[0-9]{2}-[0-9]{2}";
+    protected static String gateNumRegex = "^(\\d?[0-9]|[1-9]0)$";
+    protected String flightNumber;
+    protected String airline;
+    protected String destination;
+    protected String departureTime;
+    protected String gateNumber;
+    protected String checkIn;
+    protected String oldDepartureTime;
+
+    protected static int numOfFlights = 0;
+
+    //@@author Franky4566
+    public static void checkCommandLength(String description) throws SkyControlException {
+        if (description.isEmpty()) {
+            throw new SkyControlException(ui.showEmptyDescriptionMessage());
+        }
+    }
+
+    public static String extractDetail(String command, String start, String end) throws SkyControlException {
+
+        String extractedDetail;
+        int startIndex = command.indexOf(start) + start.length();
+        int endIndex = command.lastIndexOf(end);
+        checkNoDetailsMissing(startIndex, endIndex);
+        extractedDetail = command.substring(startIndex, endIndex).trim();
+        if (extractedDetail.equals(EMPTY_STRING)) {
+            throw new SkyControlException(ui.getMissingDetailsError());
+        }
+        return extractedDetail;
+    }
+
+    public static void checkNoDetailsMissing(int startIndex, int endIndex) throws SkyControlException {
+        if (endIndex <= startIndex || endIndex < CHECK_MISSING_DETAIL || startIndex < CHECK_MISSING_DETAIL) {
+            throw new SkyControlException(ui.getErrorMessage());
+        }
+    }
+
+    @Override
+    public void addOperation(String command) throws SkyControlException {
+        checkCommandLength(command.substring(FLIGHT_ADD_COMMAND.length()));
+        getFlightDetails(command.substring(FLIGHT_ADD_DELIMITER.length()));
+        validateDetailFormat();
+        checkFlightNumberDuplicates();
+        checkAvailableGateNumber();
+        FlightInfo flight = new FlightInfo(flightNumber, airline, destination,
+                departureTime, gateNumber, checkIn);
+        flights.add(flightIndex, flight);
+        flightIndex++;
+        ui.showFlightAddedMessage();
+    }
+
+    //@@author JordanKwua
+    @Override
+    public void deleteOperation(String detail) throws SkyControlException {
+        checkCommandLength(detail.substring(FLIGHT_DELETE_COMMAND.length()));
+        checkValidFlightNumber(detail.substring("flight delete ".length()));
+        String flightNum = detail.substring("flight delete ".length()).toUpperCase();
```

Comment:
Avoid magic strings, consider putting them into a variable.

Reply:
Also avoid complicates expression, break them into multiple steps if possible. 
Actually, perhaps you can consider hardcoding the length? you can put the length of `"flight delete "` into a constant variable like `FLIGHT_DELETE_LENGTH`.